### PR TITLE
immersed-vr: add support for darwin

### PR DESCRIPTION
**Pull request by pcasaretto**
Add support for darwin machines for the immersed-vr package.

## Description of changes

Added support for darwin machines for the immersed-vr package. I took inspiration from the 1password-gui package.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by haruki7049 — APPROVED** [review]
You have amazing skills…

File: pkgs/by-name/im/immersed-vr/package.nix [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+      hash = "sha256-CR2KylovlS7zerZIEScnadm4+ENNhib5QnS6z5Ihv1Y=";
+    };
   };
 
-  extraInstallCommands = ''
-    mv $out/bin/{${name},${pname}}
-  '';
+  src = fetchurl {
+    inherit (sources.${stdenv.system}) url hash;
+  };
```
Comment (wegank) [inline pkgs/by-name/im/immersed-vr/package.nix:33]:
```suggestion
  src = fetchurl (sources.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}"));
```

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 290893` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>immersed-vr</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 290893` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>immersed-vr</li>
  </ul>
</details>

File: pkgs/by-name/im/immersed-vr/package.nix [hunk 2]
```diff
@@ -1,27 +1,44 @@
...
-    hash = "sha256-Mx8UnV4fZSebj9ah650ZqsL/EIJpM6jl8tYmXJZiJpA=";
+  sources = {
+    x86_64-linux = {
+      url = "https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage";
+      hash = "sha256-vYk7WHGVOzrrep6vmA58ELa6aDsZFUw5D2StCYP0Ioc=";
+    };
+    aarch64-linux = {
+      url = "https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage";
+      hash = "sha256-HuhoGG2aQ2NcbZlQfUmGUl0IvhXPO5uV7x4WKJRR7Ew=";
+    };
```
Comment (wegank) [inline pkgs/by-name/im/immersed-vr/package.nix:20]:
```suggestion
    x86_64-linux = {
      url = "https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage";
      hash = "sha256-Mx8UnV4fZSebj9ah650ZqsL/EIJpM6jl8tYmXJZiJpA=";
    };
```

The hash still seems to be `sha256-Mx8UnV4fZSebj9ah650ZqsL/EIJpM6jl8tYmXJZiJpA=` here.

Comment (pcasaretto) [reply pkgs/by-name/im/immersed-vr/package.nix:20]: How does one discover these without building?
Tried:
```
> curl https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage | shasum -a 256 -b | base64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  102M  100  102M    0     0  1894k      0  0:00:55  0:00:55 --:--:--  825k
MzMxZjE0OWQ1ZTFmNjUyNzliOGZkNmExZWI5ZDE5YWFjMmZmMTA4MjY5MzNhOGU1ZjJkNjI2NWM5
NjYyMjY5MCAqLQo=
❯ nix-prefetch-url https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage
path is '/nix/store/xs5pav2cagari4rvm2zwznxy09w3h7ii-Immersed-x86_64.AppImage'
1416cab5q9nnybjshcv9h88gzhma36fyp8fniydjfr8zbsfi87rk
```

Comment (wegank) [reply pkgs/by-name/im/immersed-vr/package.nix]: https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-store-prefetch-file

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 290893` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>immersed-vr</li>
  </ul>
</details>

File: pkgs/by-name/im/immersed-vr/package.nix [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
 
-  src = fetchurl {
-    url = "https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage";
-    hash = "sha256-Mx8UnV4fZSebj9ah650ZqsL/EIJpM6jl8tYmXJZiJpA=";
+  sources = rec {
+    x86_64-linux = {
+      url = "https://web.archive.org/web/20240210075929/https://static.immersed.com/dl/Immersed-x86_64.AppImage";
+      hash = "sha256-Mx8UnV4fZSebj9ah650ZqsL/EIJpM6jl8tYmXJZiJpA=";
+    };
+    aarch64-linux = x86_64-linux;
```
Comment (wegank) [inline pkgs/by-name/im/immersed-vr/package.nix:17]:
```suggestion
```

Doesn't work (Exec format error).

**Pull request merged**